corehq/apps/locations/bulk_management.py

```diff
@@ -696,6 +703,20 @@ def _validate_types_tree(self):
                 for code in e.affected_nodes
             ]
 
+        # Verify that deleted types are not referenced by other types via foreign key
+        for deleted_type in self.types_to_be_deleted:
+            for field_name in _get_location_type_foreign_key_fields_minus_parent():
+                referencing_types_and_fields = [
+                    (lt.code, field_name) for lt in self.location_types if getattr(lt.db_object, field_name)
+                    and getattr(lt.db_object, field_name).id == deleted_type.db_object.id
+                ]
+                if referencing_types_and_fields:
+                    return [
```

Comment:
nit: FYI that this will return all location types that reference the current deleted type via the current field type, but that isn't necessarily all the conflicts, as you could have some on the other deleted types or field names.  I don't think it matters that much, just raising.  If you pull it into a helper function you could yield each conflict, then wrap the function call in `list()` to get all conflicts.

I'd preference code clarity over that particular functionality though, as I don't think this is a particularly big usability concern.